utils/vlog/tests/v2x_tests.cmake

```diff
@@ -0,0 +1,112 @@
+
+  set(V2X_GENERIC_TEST_ACTUAL_XML ${V2X_GENERIC_TEST_NAME}.${V2X_GENERIC_TEST_TYPE}.xml)
+  xml_sort(NAME ${V2X_GENERIC_TEST_NAME}_${V2X_GENERIC_TEST_TYPE}_actual_sort FILE ${V2X_GENERIC_TEST_ACTUAL_XML} OUTPUT ${V2X_GENERIC_TEST_NAME}.${V2X_GENERIC_TEST_TYPE}.actual.xml)
+
+  diff(NAME ${V2X_GENERIC_TEST_NAME}_${V2X_GENERIC_TEST_TYPE}_diff GOLDEN ${V2X_GENERIC_TEST_NAME}.${V2X_GENERIC_TEST_TYPE}.golden.xml ACTUAL ${V2X_GENERIC_TEST_NAME}.${V2X_GENERIC_TEST_TYPE}.actual.xml)
+
+  # add_dependencies(${NAME}_diff ${V2X_GENERIC_TEST_NAME}_${V2X_GENERIC_TEST_TYPE}_diff)
+  #add_dependencies(all_v2x_tests ${V2X_GENERIC_TEST_NAME}_${V2X_GENERIC_TEST_TYPE}_diff)
+
+endfunction(V2X_GENERIC_TEST)
+
+function(V2X_TEST_MODEL)
+  # ~~~
+  # V2X_TEST_MODEL(
+  #   NAME name
+  #   TOP_MODULE name
+  #   )
+  # ~~~
+  #
+  set(oneValueArgs NAME TOP_MODULE TYPE)
+  cmake_parse_arguments(
+    V2X_TEST_MODEL
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+  )
+
+  set(V2X_TEST_MODEL_SRC ${V2X_TEST_MODEL_NAME}.sim.v)
+  add_file_target(FILE ${V2X_TEST_MODEL_SRC} SCANNER_TYPE verilog)
+  v2x(NAME ${V2X_TEST_MODEL_NAME} SRCS ${V2X_TEST_MODEL_SRC} TOP_MODULE ${V2X_TEST_MODEL_TOP_MODULE})
+
+  v2x_generic_test(NAME ${V2X_TEST_MODEL_NAME} TOP_MODULE ${V2X_TEST_MODEL_TOP_MODULE} TYPE model)
+
+  add_custom_target(${V2X_TEST_MODEL_NAME}_diff ALL DEPENDS ${V2X_TEST_MODEL_NAME}_model_diff)
+endfunction(V2X_TEST_MODEL)
+
+function(V2X_TEST_PB_TYPE)
+  # ~~~
+  # V2X_TEST_PB_TYPE(
+  #   NAME name
+  #   TOP_MODULE name
+  #   )
+  # ~~~
+  #
+  set(oneValueArgs NAME TOP_MODULE TYPE)
+  cmake_parse_arguments(
+    V2X_TEST_PB_TYPE
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+  )
+
+  set(V2X_TEST_PB_TYPE_SRC ${V2X_TEST_PB_TYPE_NAME}.sim.v)
+  add_file_target(FILE ${V2X_TEST_PB_TYPE_SRC} SCANNER_TYPE verilog)
+  v2x(NAME ${V2X_TEST_PB_TYPE_NAME} SRCS ${V2X_TEST_PB_TYPE_SRC} TOP_MODULE ${V2X_TEST_PB_TYPE_TOP_MODULE})
+
+  v2x_generic_test(NAME ${V2X_TEST_PB_TYPE_NAME} TOP_MODULE ${V2X_TEST_PB_TYPE_TOP_MODULE} TYPE pb_type)
+
+  add_custom_target(${V2X_TEST_PB_TYPE_NAME}_diff ALL DEPENDS ${V2X_TEST_PB_TYPE_NAME}_pb_type_diff)
+endfunction(V2X_TEST_PB_TYPE)
+
+function(V2X_TEST_BOTH)
+  # ~~~
+  # V2X_TEST_BOTH(
+  #   NAME name
+  #   TOP_MODULE name
+  #   )
+  # ~~~
+  #
+  set(oneValueArgs NAME TOP_MODULE TYPE)
+  cmake_parse_arguments(
+    V2X_TEST_BOTH
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+  )
+
+  set(V2X_TEST_BOTH_SRC ${V2X_TEST_BOTH_NAME}.sim.v)
+  add_file_target(FILE ${V2X_TEST_BOTH_SRC} SCANNER_TYPE verilog)
+  v2x(NAME ${V2X_TEST_BOTH_NAME} SRCS ${V2X_TEST_BOTH_SRC} TOP_MODULE ${V2X_TEST_BOTH_TOP_MODULE})
+
+  v2x_generic_test(NAME ${V2X_TEST_BOTH_NAME} TOP_MODULE ${V2X_TEST_BOTH_TOP_MODULE} TYPE pb_type)
+  v2x_generic_test(NAME ${V2X_TEST_BOTH_NAME} TOP_MODULE ${V2X_TEST_BOTH_TOP_MODULE} TYPE model)
```

Comment:
Should add a ${NAME}_update_golden target which updates the golden target.

Reply:
Still need a _update_golden target

Reply:
Could you clarify on this point? These tests, as far as I understood, are made to check regressions and their purpose is to check the XML produced by `v2x` against a manually written golden `pb_type/model`. I do not think that there should be a target that updates the corresponding golden of a test.

Reply:
The target will not be an all target.  The idea being if we intentionally change v2x that affects the output , someone can run the specific update_golden target to update the test output.

Reply:
So you mean that, we update the golden automatically only if we make sure that v2x provides a correct output right?
It would be something like `make dsp_combinational_update_pb_type_golden` which will indeed update the golden `pb_type` of the dsp_combinational test, right?

Reply:
Yes.  Then the PR that updates V2X would include both a code change and diff on the output, which can then be reviewed together.

Reply:
Ok, got it, thanks. I'll add the update_golden target than, and for now I'll leave it blank as it will be filled in the following PRs

Reply:
Done